File: CS50/pset6/caesar.py
from sys import argv, exit
import cs50

# check user`s key
length = len(argv)

if length == 2:
    k = argv[1]
    if k.isdigit():
        k = int(argv[1])
        if k >= 1:
            k = int(argv[1])
        else:
            print("Usage: python caesar.py k")
            exit(1)
    else:
        print("Usage: python caesar.py k")
        exit(1)
else:
    print("Usage: python caesar.py k")
    exit(1)


# promt for user`s plaintext
plaintext = cs50.get_string("plaintext: ")

# decompose plaintext
chars = list(plaintext)

print("ciphertext: ", end="")

# convert plaintext to ciphertext
for i in range(len(chars)):
    asc = ord(chars[i])
    if asc >= 65 and asc <= 90:
        cipher = (((asc - 65) + k) % 26) + 65
        print(chr(cipher), end="")
    elif asc >= 97 and asc <= 122:
        cipher = (((asc - 97) + k) % 26) + 97
        print(chr(cipher), end="")
    else:
        print(chars[i], end="")

print("\n")

File: CS50/pset2/caesar.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>

int main(int argc, string argv[])
{    
    //check argumets, should be only 2
    if (argc != 2)
    {
        printf("Usage: ./caesar key\n");
        return 1;
    }
    
    //convert argument to integer
    int key = atoi(argv[1]);
    
    if (key < 0)
    {
        printf("Usage: ./caesar key\n");
        return 1;
    }
    
    //get text
    string p = get_string("plaintext: ");
    //print out text
    printf("ciphertext: ");
    
    //iterate through plaitext letter by letter
    for (int i = 0, n = strlen(p); i < n; i++)
    {  
        //check if this is uppercase letter
        if (p[i] >= 'A' && p[i] <= 'Z')
        {
            //%26 because in English 26 letters
            //print out uppercase letter
            printf("%c", (((p[i] - 'A') + key) % 26) + 'A');
        }
        //check if this is lowercase letter
        else if (p[i] >= 'a' && p[i] <= 'z')
        {
            //print out lowercase letter
            printf("%c", (((p[i] - 'a') + key) % 26) + 'a');
        }
        //if this symbol(not letter)/
        else
        {
            printf("%c", p[i]);
        }
    }
    printf("\n");
    return 0;
}
